src/Test/Utilities/Portable/Compilation/CompilationExtensions.cs

```diff
@@ -251,5 +251,23 @@ internal static bool CanHaveExecutableCodeBlock(ISymbol symbol)
             }
         }
 
+        public static void CheckOperations(this Compilation compilation)
+        {
+            foreach (var tree in compilation.SyntaxTrees)
+            {
+                var semanticModel = compilation.GetSemanticModel(tree);
+                var root = tree.GetRoot();
+
+                foreach (var node in root.DescendantNodesAndSelf())
+                {
+                    var operation = semanticModel.GetOperationInternal(node);
```

Comment:
We can make this faster by avoiding doing this for entire syntax tree. We can loop through all declared symbols in the compilation, then look at the declaring syntax references for each symbol and walk all the descendant syntax nodes.

Reply:
> We can make this faster by avoiding doing this for entire syntax tree.

I actually think we should do this for each node. Walking a tree is a different kind of verification.

Reply:
Alright, makes sense. We can consider the alternate approach if it becomes too slow.

Reply:
Is there value to pass all those syntax node to `GetOperation`, just as some sort of robustness check?